python/shotgun_desktop/upgrade_startup.py

```diff
@@ -48,49 +45,30 @@ def upgrade_startup(splash, sgtk, app_bootstrap):
     if not _supports_get_from_location_and_paths(sgtk):
         return False
 
-    current_desc = sgtk.deploy.descriptor.get_from_location_and_paths(
-        sgtk.deploy.descriptor.AppDescriptor.FRAMEWORK,
-        app_bootstrap.get_shotgun_desktop_cache_location(),
-        os.path.join(
-            app_bootstrap.get_shotgun_desktop_cache_location(), "install"
-        ),
+    current_desc = sgtk.descriptor.create_descriptor(
+        sgtk.get_authenticated_user().create_sg_connection(),
+        sgtk.descriptor.Descriptor.FRAMEWORK,
         get_location(sgtk, app_bootstrap)
     )
 
     # A Dev descriptor means there is nothing to update. Early out so that we don't show
     # "Getting Shotgun Desktop updates...""
-    if isinstance(current_desc, sgtk.deploy.dev_descriptor.TankDevDescriptor):
+    if current_desc.is_dev():
```

Comment:
is this check needed? won't the dev descriptor just always be up to date and therefore just do the right thing?